extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java

```diff
@@ -49,78 +41,139 @@ public static YoutubeStreamLinkHandlerFactory getInstance() {
-                    return Parser.matchGroup1("v=" + ID_PATTERN, query);
-                } catch (UnsupportedEncodingException uee) {
-                    throw new ParsingException("Could not parse attribution_link", uee);
+            if (scheme != null && scheme.equals("vnd.youtube")) {
+                String schemeSpecificPart = uri.getSchemeSpecificPart();
+                if (schemeSpecificPart.startsWith("//")) {
+                    urlString = "https:" + schemeSpecificPart;
+                } else {
+                    return assertIsID(schemeSpecificPart);
                 }
             }
-            if (url.contains("vnd.youtube")) {
-                return Parser.matchGroup1(ID_PATTERN, url);
-            }
-            if (url.contains("embed")) {
-                return Parser.matchGroup1("embed/" + ID_PATTERN, url);
-            }
-            if (url.contains("googleads")) {
-                throw new FoundAdException("Error found add: " + url);
-            }
-            return Parser.matchGroup1("[?&]v=" + ID_PATTERN, url);
+        } catch (URISyntaxException ignored) {
         }
-        if (lowercaseUrl.contains("youtu.be")) {
-            if (lowercaseUrl.contains("list=")) {
-                throw new ParsingException("Error no suitable url: " + url);
-            }
-            if (url.contains("v=")) {
-                return Parser.matchGroup1("v=" + ID_PATTERN, url);
+
+        URL url;
+        try {
+            url = Utils.stringToURL(urlString);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("The given URL is not valid");
+        }
+
+        String host = url.getHost();
+        String path = url.getPath();
+        // remove leading "/" of URL-path if URL-path is given
+        if (!path.isEmpty()) {
+            path = path.substring(1);
+        }
+
+        if (!YoutubeParsingHelper.isYoutubeALikeURL(url)) {
+            if (host.equalsIgnoreCase("googleads.g.doubleclick.net")) {
+                throw new FoundAdException("Error found ad: " + urlString);
             }
-            return Parser.matchGroup1("[Yy][Oo][Uu][Tt][Uu]\\.[Bb][Ee]/" + ID_PATTERN, url);
+
+            throw new ParsingException("The url is not a Youtube-URL");
+        }
+
+        if (YoutubePlaylistLinkHandlerFactory.getInstance().acceptUrl(urlString)) {
+            throw new ParsingException("Error no suitable url: " + urlString);
         }
-        if (lowercaseUrl.contains("hooktube")) {
-            if (lowercaseUrl.contains("&v=")
-                    || lowercaseUrl.contains("?v=")) {
-                return Parser.matchGroup1("[?&]v=" + ID_PATTERN, url);
+
+        // using uppercase instead of lowercase, because toLowercase replaces some unicode characters
+        // with their lowercase ASCII equivalent. Using toLowercase could result in faultily matching unicode urls.
+        switch (host.toUpperCase()) {
+            case "WWW.YOUTUBE-NOCOOKIE.COM": {
```

Comment:
This will require a change in the androidmanifest of the frontend.

Reply:
I will ad the url to manifest later, but originally i added this code to get throught the test:
https://github.com/connectety/NewPipeExtractor/blob/19288c1456d667644e3d0ec12bf15d3d21b88447/extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamLinkHandlerFactoryTest.java#L71
https://github.com/connectety/NewPipeExtractor/blob/19288c1456d667644e3d0ec12bf15d3d21b88447/extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeStreamLinkHandlerFactoryTest.java#L93

Reply:
all right :)

Reply:
did so in https://github.com/TeamNewPipe/NewPipe/pull/2092